packages/module/src/DataViewTableTree/DataViewTableTree.tsx

```diff
@@ -87,6 +90,33 @@ export const DataViewTableTree: FC<DataViewTableTreeProps> = ({
   const [ expandedNodeIds, setExpandedNodeIds ] = useState<string[]>([]);
   const [ expandedDetailsNodeNames, setExpandedDetailsNodeIds ] = useState<string[]>([]);
 
+  // Helper function to collect all node IDs that have children (are expandable)
+  const getExpandableNodeIds = (nodes: DataViewTrTree[]): string[] => {
+    const expandableIds: string[] = [];
+    
+    const traverse = (nodeList: DataViewTrTree[]) => {
+      nodeList.forEach(node => {
+        if (node.children && node.children.length > 0) {
+          expandableIds.push(node.id);
+          traverse(node.children);
+        }
+      });
+    };
+    
+    traverse(nodes);
+    return expandableIds;
+  };
+
+  // Effect to handle expandAll behavior
+  useEffect(() => {
+    if (expandAll) {
+      const expandableIds = getExpandableNodeIds(rows);
+      setExpandedNodeIds(expandableIds);
+    } else {
+      setExpandedNodeIds([]);
+    }
+  }, [ expandAll, rows ]);
```

Comment:
This `useEffect can be fairly complex and it can retrigger re-renders because we have `rows` in the dependency array. Let's add memoized IDs of possible expandable IDs and re-run only if that changes
```suggestion
  // Memoize the expandable IDs to avoid recalculating when rows object reference changes but structure is the same
  const expandableIds = useMemo(() => getExpandableNodeIds(rows), [ rows ]);

  // Effect to handle expandAll behavior - only runs when IDs actually change
  useEffect(() => {
    if (expandAll) {
      setExpandedNodeIds(expandableIds);
    } else {
      setExpandedNodeIds([]);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [ expandAll, expandableIds.join(',') ]);
```